doc/source/rllib-algorithms.rst

```diff
@@ -887,3 +887,61 @@ Intrinsic rewards for each env-step are calculated by taking the euclidian dista
 ("forward" model).
 This allows the agent to explore areas of the environment, where the "forward" model still performs poorly (are not "understood" yet), whereas exploration to these areas will taper down after the agent has visited them
 often: The "forward" model will eventually get better at predicting these next latent vectors, which in turn will diminish the intrinsic rewards (decrease the euclidian distance between predicted and actual vectors).
+
+
+
+.. _RE3:
+
+RE3 (Random Encoders for Efficient Exploration)
+-----------------------------------------------
+
+|Tensorflow|
+`[paper] <https://arxiv.org/pdf/2102.09430.pdf>`__
+`[implementation] <https://github.com/ray-project/ray/blob/master/rllib/utils/exploration/random_encoder.py>`__
+
+Tuned examples:
+`LunarLanderContinuous-v2 <https://github.com/ray-project/ray/blob/master/rllib/examples/re3_exploration.py>`__ (use ``--env LunarLanderContinuous-v2`` command line option)
+`Test case with Pendulum-v1 example <https://github.com/ray-project/ray/blob/master/rllib/utils/exploration/tests/test_random_encoder.py>`__
+
+**Activating RE3**
+The RE3 plugin can be easily activated by specifying it as the Exploration class to-be-used
+in the main Trainer config and inheriting the `RE3UpdateCallbacks` as shown in this `example <https://github.com/ray-project/ray/blob/c9c3f0745a9291a4de0872bdfa69e4ffdfac3657/rllib/utils/exploration/tests/test_random_encoder.py#L35>`__. Most of its parameters usually do not have to be specified as the module uses the values from the paper by default. For example:
+
+.. code-block:: python
+
+    config = sac.DEFAULT_CONFIG.copy()
+    config["env"] = "Pendulum-v1"
+    config["seed"] = 12345
+    config["callbacks"] = RE3Callbacks
+    config["exploration_config"] = {
+    	"type": "RE3",
+         # the dimensionality of the observation embedding vectors in latent space.
+         "embeds_dim": 128, 
+	 "rho": 0.1, # Beta decay factor, used for on-policy algorithm.
+         "k_nn": 50, # Number of neighbours to set for K-NN entropy estimation.
+         # Configuration for the encoder network, producing embedding vectors from observations.
+         # This can be used to configure fcnet- or conv_net setups to properly process any
+         # observation space. By default uses the Policy model configuration.
+         "encoder_net_config": {
+             "fcnet_hiddens": [],
+             "fcnet_activation": "relu",
+         },
+         "beta": 0.2 # Hyperparameter to choose between exploration and exploitation.
```

Comment:
Maybe this explanation can be a bit longer. As in what does 0.2 mean? Is that a chance of 1 in 5 on exploitation?

Reply:
Beta weights the intrinsic rewards, beta 1 means intrinsic rewards added as it is. `reward = r + beta * intrinsic_rewards`

Reply:
I have updated the doc string, please check.

Reply:
Yep, this helps. Thank you